basestar-storage/src/main/java/io/basestar/storage/util/KeysetPagingUtils.java

```diff
@@ -80,19 +75,81 @@ private KeysetPagingUtils() {
 
     public static Page.Token keysetPagingToken(final InstanceSchema schema, final List<Sort> sort, final Map<String, Object> object) {
 
-        try(final ByteArrayOutputStream baos = new ByteArrayOutputStream();
-            final DataOutputStream dos = new DataOutputStream(baos)) {
-            for (final Sort s : sort) {
-                final Name name = s.getName();
-                final Use<Object> type = schema.typeOf(name);
-                final Object value = type.create(name.get(object));
-                type.serialize(value, dos);
-            }
+        try (final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+             final DataOutputStream dos = new DataOutputStream(baos)) {
+            addSortingInfo(schema, sort, object, dos);
+            dos.flush();
+            return new Page.Token(baos.toByteArray());
+        } catch (final IOException e) {
+            // Shouldn't be possible, not doing real IO
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public static Page.Token countPreservingKeysetPagingToken(final InstanceSchema schema,
```

Comment:
Could you change this so it can wrap any kind of Page.Token rather than being coupled to the keysetPaging token implementation, this can be achieved by (de)serialising the count, then (de)serialising the (arbitrary) token bytes.